src/coreclr/scripts/jitrollingbuild.py

```diff
@@ -0,0 +1,539 @@
+            item_name = url.split("/")[-1]
+
+            if url.lower().endswith(".zip"):
+                # Delete everything in the temp_location (from previous iterations of this loop, so previous URL downloads).
+                temp_location_items = [os.path.join(temp_location, item) for item in os.listdir(temp_location)]
+                for item in temp_location_items:
+                    if os.path.isdir(item):
+                        shutil.rmtree(item)
+                    else:
+                        os.remove(item)
+
+                download_path = os.path.join(temp_location, item_name)
+
+                print("Download: {} -> {}".format(url, download_path))
+                urllib.request.urlretrieve(url, download_path)
+
+                print("Uncompress {}".format(download_path))
+                with zipfile.ZipFile(download_path, "r") as file_handle:
+                    file_handle.extractall(temp_location)
+
+                # Copy everything that was extracted to the target directory.
+                items = [ os.path.join(temp_location, item) for item in os.listdir(temp_location) if not item.endswith(".zip") ]
+                for item in items:
+                    target_path = os.path.join(target_dir, os.path.basename(item))
+                    print("Copy {} -> {}".format(item, target_path))
+                    shutil.copy2(item, target_dir)
+                    local_files.append(target_path)
+            else:
+                # Not a zip file; download directory to target directory
+                download_path = os.path.join(target_dir, item_name)
+
+                print("Download: {} -> {}".format(url, download_path))
+                urllib.request.urlretrieve(url, download_path)
+
+                local_files.append(download_path)
+
+    return local_files
+
+
+def download_command(coreclr_args):
+    """ Download the JIT
+
+    Args:
+        coreclr_args (CoreclrArguments): parsed args
+    """
+
+    print("JIT download")
+
+    jit_name = determine_jit_name(coreclr_args)
+    blob_folder_name = "{}/{}/{}/{}/{}/{}".format(az_builds_root_folder, coreclr_args.git_hash, coreclr_args.host_os, coreclr_args.arch, coreclr_args.build_type, jit_name)
+    blob_uri = "{}/{}".format(az_blob_storage_container_uri, blob_folder_name)
+    urls = [ blob_uri ]
+    download_urls(urls, coreclr_args.target_dir)
+
+
+def list_command(coreclr_args):
+    """ List the JITs in Azure Storage
+
+    Args:
+        coreclr_args (CoreclrArguments) : parsed args
+    """
+
+    blob_filter_string = "{}/{}/{}/{}".format(coreclr_args.git_hash, coreclr_args.host_os, coreclr_args.arch, coreclr_args.build_type)
+    blob_prefix_filter = "{}/{}/{}".format(az_blob_storage_container_uri, az_builds_root_folder, blob_filter_string).lower()
+
+    # Determine if a URL in Azure Storage should be allowed. The URL looks like:
+    #   https://clrjit.blob.core.windows.net/jitsuperpmi/git_hash/Linux/x64/Checked/clrjit.dll.zip
```

Comment:
We're not zipping yet, are we? So no `.zip` suffix ?

Reply:
I left the zip/unzip code in just in case we decide to use it later, but yes, this comment (and a few other copy-paste comments from superpmi.py) could be fixed. Leaving for next PR.